dbt/include/snowflake/macros/materializations/dynamic_table/ddl.sql

```diff
@@ -1,46 +1,96 @@
+    {{- log('Applying CREATE to: ' ~ dynamic_table.path) -}}
 
-    create or replace dynamic table {{ relation }}
-        lag = '{{ config.get("target_lag") }}'
-        warehouse = {{ config.get("warehouse") }}
-        as ({{ sql }})
+    create or replace dynamic table {{ dynamic_table.path }}
+        target_lag = '{{ dynamic_table.target_lag }}'
+        warehouse = {{ dynamic_table.warehouse }}
+        as ({{ dynamic_table.query }})
+    ;
+    alter dynamic table {{ dynamic_table.path }} refresh
 
 {%- endmacro %}
 
 
-{% macro snowflake__get_replace_dynamic_table_as_sql(relation, sql, existing_relation, backup_relation, intermediate_relation) -%}
-    {{- log('Applying REPLACE to: ' ~ relation) -}}
-    {{ snowflake__get_drop_dynamic_table_sql(existing_relation) }};
-    {{ snowflake__get_create_dynamic_table_as_sql(relation, sql) }}
+{% macro snowflake__replace_dynamic_table_sql(dynamic_table, existing_relation, backup_relation, intermediate_relation) -%}
+    {{- log('Applying REPLACE to: ' ~ dynamic_table.path) -}}
+
+    {{ snowflake__drop_relation_sql(existing_relation) }};
+    {{ snowflake__create_dynamic_table_sql(dynamic_table) }}
 {%- endmacro %}
 
 
-{% macro snowflake__refresh_dynamic_table(relation) -%}
-    {{- log('Applying REFRESH to: ' ~ relation) -}}
-    {%- do return('') -%}
+{% macro snowflake__refresh_dynamic_table_sql(dynamic_table) -%}
+    {{- log('Applying REFRESH to: ' ~ dynamic_table.path) -}}
+
+    alter dynamic table {{ dynamic_table.path }} refresh
 {%- endmacro %}
 
 
-{% macro snowflake__get_dynamic_table_configuration_changes(relation, new_config) -%}
-    {{- log('Determining configuration changes on: ' ~ relation) -}}
-    {%- do return(None) -%}
+{% macro snowflake__dynamic_table_configuration_changes(dynamic_table) -%}
+    {{- log('Getting CHANGES on: ' ~ dynamic_table.path) -}}
+
+    {% set existing_dynamic_table = snowflake__describe_dynamic_table(dynamic_table) %}
+    {% set _configuration_changes = this.dynamic_table_config_changeset(dynamic_table, existing_dynamic_table) %}
+    {% do return(_configuration_changes) %}
 {%- endmacro %}
 
 
-{% macro snowflake__get_drop_dynamic_table_sql(relation) %}
-    drop dynamic table if exists {{ relation }}
+{% macro snowflake__drop_dynamic_table_sql(dynamic_table) %}
+    {{- log('Applying DROP to: ' ~ dynamic_table.path) -}}
+
+    drop dynamic table if exists {{ dynamic_table.path }}
+{% endmacro %}
+
+
+{% macro snowflake__describe_dynamic_table(dynamic_table) %}
+    {{- log('Getting DESCRIBE on: ' ~ dynamic_table.path) -}}
+
+    {%- set _dynamic_table_sql -%}
+        show dynamic tables
+            like '{{ dynamic_table.name }}'
+            in schema {{ dynamic_table.schema_path }}
+        ;
+        select
+            "name",
+            "schema_name",
+            "database_name",
+            "text",
+            "target_lag",
+            "warehouse"
+        from table(result_scan(last_query_id()))
```

Comment:
concurrency nightmare possible? Maybe.....we could do a dynamic query call? Or are you confident this is thread-safe/otherwise-collision-safe?

Reply:
This was the recommended way to read the results from `show dynamic tables` according to the docs I found. I could see if I can find that again if you want. I'm not tied to it though; it worked so I went with it.